### PR TITLE
palemoon: 27.1.1 -> 27.2.1

### PR DESCRIPTION
**Pull request by calvertvl**
###### Motivation for this change
Version bump.

Also a bit of whitespace cleanup, and refactoring to use release tags.

###### Things done

- [X] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/networking/browsers/palemoon/default.nix [hunk 1]
```diff
@@ -3,20 +3,20 @@
...
-  version = "27.1.1";
+  version = "27.2.1";
 
   src = fetchFromGitHub {
     name   = "palemoon-src";
     owner  = "MoonchildProductions";
     repo   = "Pale-Moon";
-    rev    = "a35936746069e6591181eb67e5f9ea094938bae5";
+    rev    = version + "_Release";
     sha256 = "0hns5993dh93brwz3z4xp1zp8n90x1hajxylv17zybpysax64jsk";
```
Comment (RonnyPfannschmidt) [inline pkgs/applications/networking/browsers/palemoon/default.nix]: won't this bit break?

**Comment by rnhmjoj** [comment]
While you are it can you fix the desktop entry?
I forgot to add
```
mkdir -p $out/share/applications
cp ${desktopItem}/share/applications/* $out/share/applications
```
to the install phase.

File: pkgs/applications/networking/browsers/palemoon/default.nix (hunk 1 above)
Comment (rnhmjoj) [inline pkgs/applications/networking/browsers/palemoon/default.nix]: No, it's fine. See how the release are tagged: https://github.com/MoonchildProductions/Pale-Moon/releases

Comment (RonnyPfannschmidt) [inline pkgs/applications/networking/browsers/palemoon/default.nix]: @rnhmjoj i meant the unchanged content hash making it confuse wrt the fixed output derivation

Comment (calvertvl) [inline pkgs/applications/networking/browsers/palemoon/default.nix]: The version+"_Release" bit builds the tag name; these can be used when fetching from git.

It's also a bit more obvious than just using a hash.

**Comment by calvertvl** [comment]
Pushed the updated version, with the desktop entry fix from @rnhmjoj; it actually picks up the correct version of the source code this time.

**Review by rnhmjoj — APPROVED** [review]

**Pull request merged**